app/routes/companyInterest/components/CompanyInterestPage.tsx

```diff
@@ -899,215 +945,32 @@ const CompanyInterestPage = (props: Props) => {
-                      <h4>{EVENT_TYPES.other[language]}</h4>
-                      <p className={styles.label}>*</p>
-                    </Flex>
-                    <p>{interestText.otherEventDescription[language]}</p>
-                    <Field
-                      placeholder={interestText.otherEventComment[language]}
-                      name="otherEventComment"
-                      component={TextEditor.Field}
-                      rows={10}
-                      className={styles.textEditor}
-                    />
-                  </div>
-                )
-              );
-            })}
-
-            {spyValues((values: CompanyInterestFormEntity) => {
-              const showStartupComment = values.events?.some(
-                (e) => e.name === 'start_up' && e.checked === true
-              );
-
-              return (
-                showStartupComment && (
-                  <div className={styles.topline}>
-                    <Flex alignItems="center" gap={1}>
-                      <h4>{EVENT_TYPES.start_up[language]}</h4>
-                      <p className={styles.label}>*</p>
-                    </Flex>
-                    <p>{interestText.startUpDescription[language]}</p>
-                    <Field
-                      placeholder={interestText.startUpComment[language]}
-                      name="startupComment"
-                      component={TextEditor.Field}
-                      rows={10}
-                      className={styles.textEditor}
-                    />
-                  </div>
-                )
-              );
-            })}
-
-            {spyValues((values: CompanyInterestFormEntity) => {
-              const showCompanyToCompanyComment = values.events?.some(
-                (e) => e.name === 'company_to_company' && e.checked === true
-              );
-
-              return (
-                showCompanyToCompanyComment && (
-                  <div className={styles.topline}>
-                    <Flex alignItems="center" gap={1}>
-                      <h4>{EVENT_TYPES.company_to_company[language]}</h4>
-                      <p className={styles.label}>*</p>
-                    </Flex>
-
-                    <p>{interestText.companyToCompanyDescription[language]}</p>
-                    <Field
-                      placeholder={
-                        interestText.companyToCompanyComment[language]
-                      }
-                      name="companyToCompanyComment"
-                      component={TextEditor.Field}
-                      rows={10}
-                      className={styles.textEditor}
-                    />
-                  </div>
-                )
-              );
+            {eventCommentFields.map((event) => {
+              return spyValues((values: CompanyInterestFormEntity) => {
+                const showComment = values.events?.some(
+                  (e) => e.name === event.name && e.checked === true
+                );
```

Comment:
Haven't looked into how these values work exactly, so I'm just throwing it out there so you can decide if it makes sense.

As you're checking the array for events that are .checked (I assume that you have covered all the event names in the array above - so that the `e.name === event.name` check would only make sure that specific event is checked), could it be an option to instead do something like the following?
```javascript
spyValues((values: CompanyInterestFormEntity) => 
    values.events?.filter(event => event.checked).map(event => {
        const eventFields = eventCommentFields[event.name]
        return (...);
    })
)
```
For it to make sense, `eventCommentFields` would need to be an object with what is currently the `name: ""` value as the key.

Just an alternative approach to do the same thing, but it would shorten it down a little bit more, and remove the need for the `some()` calls and boolean values making it a little bit more readable(:

Reply:
➕